src/hotspot/share/runtime/frame.cpp

```diff
@@ -1558,6 +1558,39 @@ void frame::describe(FrameValues& values, int frame_no, const RegisterMap* reg_m
 
 #endif
 
+/**
+ * Gets the caller frame of `fr` for thread `t`.
+ *
+ * @returns an invalid frame (i.e. fr.pc() === 0) if the caller cannot be obtained
+ */
+frame frame::next_frame(frame fr, Thread* t) {
```

Comment:
`frame` is a non-trivial type to copy, so it seems like it would be better to pass it by const-ref rather
than by value.  That might trip over lots of const-correctness issues when accessing it though.  If
pulling that thread looks like an excessive tangle then clean this up as a followup.

Reply:
It seems we often pass frames around rather than pointers or references to them, but I don't know why. So I would not want to be changing any existing signatures in case there is some nuance I am not aware of.